### PR TITLE
Upgrade scala version.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Also updates scalaz, scalacheck, specs2.

I found I needed to explicitly add scalar-stream to the dependencies plus add another resolver so that it actually found it. I grabbed this info from the scala-stream [github](http://github.com/scalaz/scalaz-stream) 

Also the `-Ywarn-all` seems to have been removed, so I used the closest thing I could see in the scalar options `-Xfatal-warnings`

### CODE REVIEW

**Comment by charleso** [comment]
Thanks again @tmcgilchrist. I'll merge this now, but we might also want to bump the sbt version in `project/build.properties` at some stage too.

File: build.sbt [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+  "oss snapshots"       at "http://oss.sonatype.org/content/repositories/snapshots"
+, "oss releases"        at "http://oss.sonatype.org/content/repositories/releases"
+, "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases"
 )
 
 scalacOptions := Seq(
   "-deprecation"
 , "-unchecked"
-, "-Ywarn-all"
+, "-Xfatal-warnings"
```
Comment (markhibberd) [inline build.sbt]: This is good. For reference `-Xlint` and `-Ywarn-all` became equivalent, so they dropped warn-all.

**Comment by markhibberd** [comment]
Thanks.

File: build.sbt (hunk 1 above)
Comment (tmcgilchrist) [inline build.sbt]: Good, I'm usually in favour of warnings becoming errors. Especially with things like in exhaustive pattern matches.

**Pull request merged**